pkg/datastore.go

```diff
@@ -185,6 +175,28 @@ func (s DefaultDatastore) GetRegistration(ctx context.Context, pk PubKey) (types
 	return registration, err
 }
 
+func (s DefaultDatastore) queryToKey(ctx context.Context, query HeaderQuery) (ds.Key, error) {
+	var (
+		rawKey []byte
+		err    error
+	)
+
+	if strings.Compare(query.BlockHash.String(), "0x0000000000000000000000000000000000000000000000000000000000000000") != 0 {
```

Comment:
This should go in some kid of `Validate()` method (or similar) in the `HeaderQuery` type.  Same with the other checks.